Sources/CorePayments/Eligibility/EligibilityAPI.swift

```diff
@@ -1,16 +1,75 @@
 import Foundation
 
-final class EligibilityAPI {
+/// API that return merchant's eligibility for different payment methods: Venmo, PayPal, PayPal Credit, Pay Later & credit card
+class EligibilityAPI {
     
     // MARK: - Private Propertires
     
     private let coreConfig: CoreConfig
     private let networkingClient: NetworkingClient
     
-    // MARK: - Initializer
+    // MARK: - Initializers
     
+    /// Initialize the eligibility API to check for payment methods eligibility
+    /// - Parameter coreConfig: configuration object
     init(coreConfig: CoreConfig) {
         self.coreConfig = coreConfig
         self.networkingClient = NetworkingClient(coreConfig: coreConfig)
     }
+    
+    /// Exposed for injecting MockNetworkingClient in tests
+    init(coreConfig: CoreConfig, networkingClient: NetworkingClient) {
+        self.coreConfig = coreConfig
+        self.networkingClient = networkingClient
+    }
+    
+    // MARK: - Internal Methods
+    
+    /// Checks merchants eligibility for different payment methods.
+    /// - Returns: An `EligibilityResponse` containing the result of the eligibility check.
+    func check(_ eligibilityRequest: EligibilityRequest) async throws -> EligibilityResponse {
+        let variables = EligibilityVariables(eligibilityRequest: eligibilityRequest, clientID: coreConfig.clientID)
+        let graphQLRequest = GraphQLRequest(query: Self.rawQuery, variables: variables)
+        let httpResponse = try await networkingClient.fetch(request: graphQLRequest)
+        
+        return try HTTPResponseParser().parseGraphQL(httpResponse, as: EligibilityResponse.self)
+    }
+}
+
+extension EligibilityAPI {
+    
+    static let rawQuery = """
```

Comment:
I don't think we really need this to be in an extension here. Was there a particular reason for using it?

Reply:
I decided to put the query in an extension to make the code more readable. I based it on [this](https://github.com/paypal/paypal-ios/blob/1eb22f17fe1244d0674c1b5d3552caa97214e9dc/Sources/CardPayments/APIRequests/VaultPaymentTokensAPI.swift#L31) implementation where the query is in the same method, which makes the method much larger. Basically, it's to separate responsibilities (ish)

Reply:
I decided to put the query in an extension to make the code more readable. I based it on [this](https://github.com/paypal/paypal-ios/blob/1eb22f17fe1244d0674c1b5d3552caa97214e9dc/Sources/CardPayments/APIRequests/VaultPaymentTokensAPI.swift#L31) implementation where the query is in the same method, which makes the method much larger. Basically, it's to separate(ish) responsibilities

Reply:
We could also throw it in a private method under `check` if we want to separate it but move it out of the extension